### PR TITLE
Add mvc

### PR DESCRIPTION
**Pull request by ty19**
Added Model View Controller (MVC) architecture to the GUI. Also added comments. This should make things easier to understand, and keep things clean when we add more features.

We can't really see the usefulness of MVC with what we're doing since we aren't going to rewrite the backend or GUI anytime soon. I guess it's nice to have, even if it's not all that beneficial. 

Side note, line counts are off since git can't tell that I split gui_final.py into two files. Not sure if there's a command like git mv or something to have it detect this.

### CODE REVIEW

**Comment by rjcrjc** [comment]
TFW you take comp 310 and feel like you have to apply the material somewhere.

**Comment by ty19** [comment]
Yeah pretty much

File: view.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def init_mission_control_tab(self):
         tk.ttk.Button(valve_frame, text="Set Valve", command=lambda: self.backend_adapter.send(ServerInfo.SET_VALVE))\
             .grid(row=1, column=1, padx=15, pady=10)
 
-        tk.ttk.Button(valve_frame, text="Unset Valve", command=lambda: self.backend_adapter.send(ServerInfo.UNSET_VALVE))\
+        tk.ttk.Button(valve_frame, text="Unset Valve", command=lambda: self.backend_adapter.send(ServerInfo.SET_VALVE))\
```
Comment (rjcrjc) [inline view.py]: I'm really confused about this.

Comment (rjcrjc) [reply view.py]: I think that might have been due to some other bug???

**Comment by rjcrjc** [comment]
Looking through these:
hard to verify, but I assume that the only major change with the first commit is splitting up gui_final into the two separate files...

third commit looks like a mix of mvc stuff and also some gui stuff that's tkinter specific that I'm not really qualified to evaluate, but I bet it's fine. I think it's funny that the initial backend adapter looks like it's just the literal backend. Don't tell swong.

In 4th commit you replace backend adapter with something more proper. Also I think it might be cool/nicer to make the methods that get the queue into just a queue member object that's a reference to the queue in the backend, that way the old semantics of backend_adapter.queues would still work. Maybe that's bad?

The rest of it looks fine other than a question I have about one of the button labels.

I haven't actually run this code or tested it in any way.

**Review by rjcrjc — APPROVED** [review]
lgtm

**Pull request merged**